jetbrains-core/src/software/aws/toolkits/jetbrains/ui/feedback/FeedbackDialog.kt

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.feedback
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.ValidationInfo
+import icons.AwsIcons
+import org.jetbrains.annotations.TestOnly
+import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
+import software.aws.toolkits.resources.message
+
+class FeedbackDialog(project: Project) : DialogWrapper(project) {
+    val panel = SubmitFeedbackPanel(initiallyPositive = true)
+
+    init {
+        title = feedbackTitle
+        init()
+    }
+
+    override fun doOKAction() {
+        if (okAction.isEnabled) {
+            setOKButtonText(message("feedback.submitting"))
+            isOKActionEnabled = false
+
+            val sentiment = panel.sentiment ?: throw RuntimeException("sentiment was null after validation")
+            val comment = panel.comment ?: throw RuntimeException("comment was null after validation")
```

Comment:
Can you throw something more specific than RuntimeException? `InvalidStateException` or a custom would be more informative.